pytest_operator/plugin.py

```diff
@@ -347,46 +354,158 @@ def download_resource(self, resource, destination: Path):
         return local_file
 
 
+class ModelNotFoundError(Exception):
+    """Raise when referencing to a model that doesn't exist."""
+
+
+class ModelInUseError(Exception):
+    """Raise when trying to add a model alias which already exists."""
+
+
+@dataclasses.dataclass
+class ModelState:
+    model: Model
+    keep: bool
+    controller_name: str
+    cloud_name: Optional[str]
+    model_name: str
+    config: dict = None
+    tmp_path: Optional[Path] = None
+
+    @property
+    def full_name(self) -> str:
+        return f"{self.controller_name}:{self.model_name}"
```

Comment:
stateful storage within the `ops_test` object.  This isn't adjustable by users of `ops_test`